### PR TITLE
Fix dashboard not applying mounting point when linking or redirecting to another page

### PR DESCRIPTION
**Pull request by cyrilchandelier**
TL;DR: single app redirection is not the root of our issue here, all mounted dashboards seem broken when using 1.3.0.

When upgrading to the latest version of the dashboard, I encountered the same issue as folks in #1054: app redirects and doesn't keep the mounting point.

Since I made the single app redirection changes in the first place (#958) and it was just a couple lines of code, I started investigating and I'm now 100% confident the problem was misattributed to the single app redirection in the first place.

Unless I'm missing something, something changed between 1.2.0 and 1.3.0. Commenting the code I wrote in #958 didn't fix the issue. I tracked the bad redirection to the following line:
```
// Dashboard.js#312
<Redirect from='/' to='/apps' />`
```

But that's not it! Basically, it looks like the mounting point is lost and the navigation is totally broken when the mountPath is not `/`. I've no idea how it was even working before since the whole app is referring/redirecting to `/apps` everywhere, without referencing the eventual mount path.

This PR is the result of a painful work trying to add the mountPath everywhere I found it was needed, but to be honest, I had no idea was I was doing the whole time, there might be a simple solution.

@flovilmart: I would love your assistance and opinion on this, thanks!

### CODE REVIEW

**Comment by flovilmart** [comment]
I believe the problem is somewhere else and that rewriting it all was not necessary. I may be wrong. I don’t maintain this project anymore, so check with @acinader, he may be able to help

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by acinader** [comment]
@dplewis would you be willing to look at this?  I don't have the problem...

**Comment by dplewis** [comment]
@acinader Just looked into it. A lot of dependencies were updated. The problem is in the `history` package. You have to set a basename (base url / mountpath) to fix this issue.

I can do a separate PR with a sample server for testing.

@cyrilchandelier Thanks for getting started on this, I know it must have taken a lot of time.

**Comment by dplewis** [comment]
Superced via https://github.com/parse-community/parse-dashboard/pull/1070

**Comment by GoMino** [comment]
Can we hope for a new release with the fix? @dplewis

**Comment by TomWFox** [comment]
@GoMino I've opened a PR for a new release - #1083
